core/frontend/src/tile/map/ImageryTileTree.ts

```diff
@@ -333,12 +347,46 @@ class ImageryMapLayerTreeSupplier implements TileTreeSupplier {
           if (0 === cmp) {
             cmp = compareBooleans(lhs.settings.transparentBackground, rhs.settings.transparentBackground);
             if (0 === cmp) {
-              cmp = compareNumbers(lhs.settings.subLayers.length, rhs.settings.subLayers.length);
+              if (lhs.settings.properties || rhs.settings.properties) {
+                if (lhs.settings.properties && rhs.settings.properties) {
+                  const lhsKeysLength = Object.keys(lhs.settings.properties).length;
+                  const rhsKeysLength = Object.keys(rhs.settings.properties).length;
+
+                  if (lhsKeysLength !== rhsKeysLength) {
+                    cmp = lhsKeysLength - rhsKeysLength;
+                  } else {
+                    for (const key of Object.keys(lhs.settings.properties)) {
+                      const lhsProp = lhs.settings.properties[key];
+                      const rhsProp = rhs.settings.properties[key];
+                      if (typeof lhsProp !== typeof rhsProp) {
+                        cmp = 1;
+                        break;
+                      }
+                      if (Array.isArray(lhsProp) || Array.isArray(rhsProp)) {
+                        cmp = compareSimpleArrays(lhsProp as (number | string | boolean)[], rhsProp as (number | string | boolean)[]);

```

Comment:
Your cast is to a homogeneous array but MapLayerProperties and compareSimpleArrays operate on heterogeneous arrays. Which did you intend?